// bizcharts/lib/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  Util: true,
  Annotation: true,
  G2: true,
  GComponents: true,
  Chart: true,
  View: true,
  Tooltip: true,
  Legend: true,
  Coordinate: true,
  Axis: true,
  Facet: true,
  Slider: true,
  Area: true,
  Edge: true,
  Heatmap: true,
  Interval: true,
  Line: true,
  Point: true,
  Polygon: true,
  Schema: true,
  BaseGeom: true,
  Label: true,
  Path: true,
  LineAdvance: true,
  Geom: true,
  Coord: true,
  Guide: true,
  Effects: true,
  Interaction: true,
  createPlot: true,
  createTooltipConnector: true,
  useView: true,
  useRootChart: true,
  useChartInstance: true,
  useTheme: true,
  withView: true,
  withChartInstance: true
};
Object.defineProperty(exports, "Chart", {
  enumerable: true,
  get: function get() {
    return _Chart["default"];
  }
});
Object.defineProperty(exports, "View", {
  enumerable: true,
  get: function get() {
    return _View["default"];
  }
});
Object.defineProperty(exports, "Tooltip", {
  enumerable: true,
  get: function get() {
    return _Tooltip["default"];
  }
});
Object.defineProperty(exports, "Legend", {
  enumerable: true,
  get: function get() {
    return _Legend["default"];
  }
});
Object.defineProperty(exports, "Coordinate", {
  enumerable: true,
  get: function get() {
    return _Coordinate["default"];
  }
});
Object.defineProperty(exports, "Axis", {
  enumerable: true,
  get: function get() {
    return _Axis["default"];
  }
});
Object.defineProperty(exports, "Facet", {
  enumerable: true,
  get: function get() {
    return _Facet["default"];
  }
});
Object.defineProperty(exports, "Slider", {
  enumerable: true,
  get: function get() {
    return _Slider["default"];
  }
});
Object.defineProperty(exports, "Area", {
  enumerable: true,
  get: function get() {
    return _Area["default"];
  }
});
Object.defineProperty(exports, "Edge", {
  enumerable: true,
  get: function get() {
    return _Edge["default"];
  }
});
Object.defineProperty(exports, "Heatmap", {
  enumerable: true,
  get: function get() {
    return _Heatmap["default"];
  }
});
Object.defineProperty(exports, "Interval", {
  enumerable: true,
  get: function get() {
    return _Interval["default"];
  }
});
Object.defineProperty(exports, "Line", {
  enumerable: true,
  get: function get() {
    return _Line["default"];
  }
});
Object.defineProperty(exports, "Point", {
  enumerable: true,
  get: function get() {
    return _Point["default"];
  }
});
Object.defineProperty(exports, "Polygon", {
  enumerable: true,
  get: function get() {
    return _Polygon["default"];
  }
});
Object.defineProperty(exports, "Schema", {
  enumerable: true,
  get: function get() {
    return _Schema["default"];
  }
});
Object.defineProperty(exports, "BaseGeom", {
  enumerable: true,
  get: function get() {
    return _Base["default"];
  }
});
Object.defineProperty(exports, "Label", {
  enumerable: true,
  get: function get() {
    return _Label["default"];
  }
});
Object.defineProperty(exports, "Path", {
  enumerable: true,
  get: function get() {
    return _Path["default"];
  }
});
Object.defineProperty(exports, "LineAdvance", {
  enumerable: true,
  get: function get() {
    return _LineAdvance["default"];
  }
});
Object.defineProperty(exports, "Geom", {
  enumerable: true,
  get: function get() {
    return _geometry["default"];
  }
});
Object.defineProperty(exports, "Coord", {
  enumerable: true,
  get: function get() {
    return _coord["default"];
  }
});
Object.defineProperty(exports, "Guide", {
  enumerable: true,
  get: function get() {
    return _Guide["default"];
  }
});
Object.defineProperty(exports, "Effects", {
  enumerable: true,
  get: function get() {
    return _Effects["default"];
  }
});
Object.defineProperty(exports, "Interaction", {
  enumerable: true,
  get: function get() {
    return _Interaction["default"];
  }
});
Object.defineProperty(exports, "createPlot", {
  enumerable: true,
  get: function get() {
    return _createPlot["default"];
  }
});
Object.defineProperty(exports, "createTooltipConnector", {
  enumerable: true,
  get: function get() {
    return _createTooltipConnector["default"];
  }
});
Object.defineProperty(exports, "useView", {
  enumerable: true,
  get: function get() {
    return _useChartView["default"];
  }
});
Object.defineProperty(exports, "useRootChart", {
  enumerable: true,
  get: function get() {
    return _useChartInstance["default"];
  }
});
Object.defineProperty(exports, "useChartInstance", {
  enumerable: true,
  get: function get() {
    return _useChartInstance["default"];
  }
});
Object.defineProperty(exports, "useTheme", {
  enumerable: true,
  get: function get() {
    return _useTheme["default"];
  }
});
Object.defineProperty(exports, "withView", {
  enumerable: true,
  get: function get() {
    return _view.withView;
  }
});
Object.defineProperty(exports, "withChartInstance", {
  enumerable: true,
  get: function get() {
    return _root.withChartInstance;
  }
});
exports.GComponents = exports.G2 = exports.Annotation = exports.Util = void 0;

var _object = _interopRequireDefault(require("object.assign"));

var antUtils = _interopRequireWildcard(require("@antv/util"));

var Annotation = _interopRequireWildcard(require("./components/Annotation"));

exports.Annotation = Annotation;

var G2 = _interopRequireWildcard(require("./g2-all"));

exports.G2 = G2;

var GComponents = _interopRequireWildcard(require("./g-components"));

exports.GComponents = GComponents;

var bxUtil = _interopRequireWildcard(require("./utils"));

var _Chart = _interopRequireDefault(require("./components/Chart"));

var _View = _interopRequireDefault(require("./components/View"));

var _Tooltip = _interopRequireDefault(require("./components/Tooltip"));

var _Legend = _interopRequireDefault(require("./components/Legend"));

var _Coordinate = _interopRequireDefault(require("./components/Coordinate"));

var _Axis = _interopRequireDefault(require("./components/Axis"));

var _Facet = _interopRequireDefault(require("./components/Facet"));

var _Slider = _interopRequireDefault(require("./components/Slider"));

var _Area = _interopRequireDefault(require("./geometry/Area"));

var _Edge = _interopRequireDefault(require("./geometry/Edge"));

var _Heatmap = _interopRequireDefault(require("./geometry/Heatmap"));

var _Interval = _interopRequireDefault(require("./geometry/Interval"));

var _Line = _interopRequireDefault(require("./geometry/Line"));

var _Point = _interopRequireDefault(require("./geometry/Point"));

var _Polygon = _interopRequireDefault(require("./geometry/Polygon"));

var _Schema = _interopRequireDefault(require("./geometry/Schema"));

var _Base = _interopRequireDefault(require("./geometry/Base"));

var _Label = _interopRequireDefault(require("./geometry/Label"));

var _Path = _interopRequireDefault(require("./geometry/Path"));

var _LineAdvance = _interopRequireDefault(require("./geometry/LineAdvance"));

var _geometry = _interopRequireDefault(require("./geometry"));

var _coord = _interopRequireDefault(require("./components/Coordinate/coord"));

var _Guide = _interopRequireDefault(require("./adapter/Guide"));

var _Effects = _interopRequireDefault(require("./components/Effects"));

var _Interaction = _interopRequireDefault(require("./components/Interaction"));

var _createPlot = _interopRequireDefault(require("./createPlot"));

var _createTooltipConnector = _interopRequireDefault(require("./connector/createTooltipConnector"));

var _useChartView = _interopRequireDefault(require("./hooks/useChartView"));

var _useChartInstance = _interopRequireDefault(require("./hooks/useChartInstance"));

var _useTheme = _interopRequireDefault(require("./hooks/useTheme"));

var _view = require("./context/view");

var _root = require("./context/root");

var _core = require("./core");

Object.keys(_core).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _core[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _core[key];
    }
  });
});

var _plots = require("./plots");

Object.keys(_plots).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _plots[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _plots[key];
    }
  });
});
var Util = (0, _object["default"])((0, _object["default"])((0, _object["default"])({}, antUtils), bxUtil), G2.Util); // 基础组件
// g-components

exports.Util = Util;
