// bizcharts/lib/components/View/viewHelper.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _object = _interopRequireDefault(require("object.assign"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _isArray2 = _interopRequireDefault(require("@antv/util/lib/is-array"));

var _deepMix2 = _interopRequireDefault(require("@antv/util/lib/deep-mix"));

var _each2 = _interopRequireDefault(require("@antv/util/lib/each"));

var _warning = _interopRequireDefault(require("../../utils/warning"));

var _shallowEqual = _interopRequireDefault(require("../../utils/shallowEqual"));

var __rest = void 0 && (void 0).__rest || function (s, e) {
  var t = {};

  for (var p in s) {
    if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0) t[p] = s[p];
  }

  if (s != null && typeof Object.getOwnPropertySymbols === "function") for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
    if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i])) t[p[i]] = s[p[i]];
  }
  return t;
};

var ViewHelper = /*#__PURE__*/function () {
  function ViewHelper(chart) {
    (0, _classCallCheck2["default"])(this, ViewHelper);
    this.chart = chart;
    this.config = {};
    this.isRootView = false;
    this.chart = chart;
  }

  (0, _createClass2["default"])(ViewHelper, [{
    key: "creatViewInstance",
    value: function creatViewInstance(options) {
      this.view = this.chart.createView(this.processOptions(options)); // @ts-ignore

      this.view.rootChart = this.chart;
    }
  }, {
    key: "getView",
    value: function getView() {
      return this.view;
    }
  }, {
    key: "update",
    value: function update(newConfig) {
      var _this = this;

      // 不需要重建实例
      var preData = this.config.data;
      var scale = newConfig.scale,
          animate = newConfig.animate,
          filter = newConfig.filter,
          visible = newConfig.visible;
      var _newConfig$data = newConfig.data,
          data = _newConfig$data === void 0 ? [] : _newConfig$data;

      if (data.rows) {
        (0, _warning["default"])(!data.rows, 'bizcharts@4不支持 dataset数据格式，请使用data={dv.rows}');
        data = data.rows;
      }

      if (!this.view || (0, _isArray2["default"])(preData) && preData.length === 0) {
        // hack g2 数据切换的问题
        this.destroy();
        this.creatViewInstance(newConfig);
      } // 数据


      if ((0, _isArray2["default"])(preData)) {
        this.view.changeData(data); // 数据只做2级浅比较

        var isEqual = true;

        if (preData.length !== data.length) {
          isEqual = false;
        } else {
          preData.forEach(function (element, index) {
            if (!(0, _shallowEqual["default"])(element, data[index])) {
              isEqual = false;
            }
          });
        }

        if (!isEqual) {
          this.view.changeData(data);
        }
      } else {
        this.view.data(data);
      } // 比例尺


      this.view.scale(scale); // animate

      this.view.animate(animate); // filter

      (0, _each2["default"])(this.config.filter, function (it, index) {
        // 销毁
        if ((0, _isArray2["default"])(it)) {
          _this.view.filter(it[0], null);
        } else {
          _this.view.filter(index, null);
        }
      });
      (0, _each2["default"])(filter, function (it, index) {
        if ((0, _isArray2["default"])(it)) {
          _this.view.filter(it[0], it[1]);
        } else {
          _this.view.filter(index, it);
        }
      }); // visible 

      if (visible) {
        this.view.show();
      } else {
        this.view.hide();
      }

      this.config = (0, _object["default"])((0, _object["default"])({}, newConfig), {
        data: data
      });
    }
  }, {
    key: "destroy",
    value: function destroy() {
      if (this.view) {
        this.view.destroy();
        this.view = null;
      }

      this.config = {};
    }
  }, {
    key: "processOptions",
    value: function processOptions(options) {
      var region = options.region,
          start = options.start,
          end = options.end,
          other = __rest(options, ["region", "start", "end"]);

      (0, _warning["default"])(!start, 'start 属性将在4.1后废弃，请使用 region={{ start: {x:0,y:0}}} 替代');
      (0, _warning["default"])(!end, 'end 属性将在4.1后废弃，请使用 region={{ end: {x:0,y:0}}} 替代');
      var regionCfg = (0, _deepMix2["default"])({
        start: {
          x: 0,
          y: 0
        },
        end: {
          x: 1,
          y: 1
        }
      }, {
        start: start,
        end: end
      }, region);
      return (0, _object["default"])((0, _object["default"])({}, other), {
        region: regionCfg
      });
    }
  }]);
  return ViewHelper;
}();

exports["default"] = ViewHelper;
