// bizcharts/es/g-components/Base/Helper.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _forIn = _interopRequireDefault(require("@antv/util/lib/for-in"));

var _isFunction = _interopRequireDefault(require("@antv/util/lib/is-function"));

var _pickWithout = _interopRequireDefault(require("../../utils/pickWithout"));

var _constant = require("../../utils/constant");

var _cloneDeep = _interopRequireDefault(require("../../utils/cloneDeep"));

var _events = require("./events");

var __rest = void 0 && (void 0).__rest || function (s, e) {
  var t = {};

  for (var p in s) {
    if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0) t[p] = s[p];
  }

  if (s != null && typeof Object.getOwnPropertySymbols === "function") for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
    if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i])) t[p[i]] = s[p[i]];
  }
  return t;
};

var Helper = /*#__PURE__*/function () {
  function Helper(shape) {
    (0, _classCallCheck2["default"])(this, Helper);
    this.shape = shape;
    this.shape = shape;
  }

  (0, _createClass2["default"])(Helper, [{
    key: "createInstance",
    value: function createInstance(props) {
      this.instance = props.group.addShape(this.shape, (0, _pickWithout["default"])(props, ['group', 'ctx']));
    }
  }, {
    key: "destroy",
    value: function destroy() {
      if (this.instance) {
        this.instance.remove(true);
        this.instance = null;
      }
    }
  }, {
    key: "update",
    value: function update(props) {
      var _this = this;

      var newConfig = (0, _pickWithout["default"])(props, (0, _toConsumableArray2["default"])(_constant.REACT_PIVATE_PROPS));
      this.destroy();
      this.createInstance(newConfig);

      var attrs = newConfig.attrs,
          animate = newConfig.animate,
          isClipShape = newConfig.isClipShape,
          visible = newConfig.visible,
          matrix = newConfig.matrix,
          others = __rest(newConfig, ["attrs", "animate", "isClipShape", "visible", "matrix"]);

      this.instance.attr(attrs);

      if (animate) {
        var toAttrs = animate.toAttrs,
            animateCfg = __rest(animate, ["toAttrs"]);

        this.instance.animate(toAttrs, animateCfg);
      }

      if (isClipShape) {
        this.instance.isClipShape();
      }

      if (visible === false) {
        this.instance.hide();
      }

      if (matrix) {
        this.instance.setMatrix(matrix);
      }

      (0, _forIn["default"])(_events.EVENTS, function (v, k) {
        if ((0, _isFunction["default"])(others[k])) {
          _this.instance.on(v, others[k]);
        }
      });
      this.config = (0, _cloneDeep["default"])(newConfig);
    }
  }]);
  return Helper;
}();

exports["default"] = Helper;
