// bizcharts/lib/g-components/Group.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _react = _interopRequireDefault(require("react"));

var _forIn = _interopRequireDefault(require("@antv/util/lib/for-in"));

var _isFunction = _interopRequireDefault(require("@antv/util/lib/is-function"));

var _debounce = _interopRequireDefault(require("@antv/util/lib/debounce"));

var _isArray = _interopRequireDefault(require("@antv/util/lib/is-array"));

var _uniqueId = _interopRequireDefault(require("@antv/util/lib/unique-id"));

var _group2 = _interopRequireWildcard(require("../context/group"));

var _events = require("./Base/events");

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

var __rest = void 0 && (void 0).__rest || function (s, e) {
  var t = {};

  for (var p in s) {
    if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0) t[p] = s[p];
  }

  if (s != null && typeof Object.getOwnPropertySymbols === "function") for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
    if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i])) t[p[i]] = s[p[i]];
  }
  return t;
};

var Group = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(Group, _React$Component);

  var _super = _createSuper(Group);

  function Group(props) {
    var _this;

    (0, _classCallCheck2["default"])(this, Group);
    _this = _super.call(this, props);
    _this.state = {
      isReady: false
    };
    _this.handleRender = (0, _debounce["default"])(function () {
      if (!_this.instance) {
        var _this$props = _this.props,
            _group = _this$props.group,
            _zIndex = _this$props.zIndex,
            _name = _this$props.name; // children.push 中push 找不到

        _this.instance = _group.chart.canvas.addGroup({
          zIndex: _zIndex,
          name: _name
        });

        _group.chart.canvas.sort();

        _this.setState({
          isReady: true
        });
      } else {
        _this.forceUpdate();
      }
    }, 300);

    _this.configGroup = function (props) {
      var rotate = props.rotate,
          animate = props.animate,
          rotateAtPoint = props.rotateAtPoint,
          scale = props.scale,
          translate = props.translate,
          move = props.move;

      if (rotate) {
        _this.instance.rotate(rotate);
      }

      if ((0, _isArray["default"])(rotateAtPoint)) {
        var _this$instance;

        // @ts-ignore
        (_this$instance = _this.instance).rotateAtPoint.apply(_this$instance, (0, _toConsumableArray2["default"])(rotateAtPoint));
      }

      if (scale) {
        _this.instance.rotate(scale);
      }

      if (translate) {
        _this.instance.translate(translate[0], translate[1]);
      }

      if (move) {
        _this.instance.move(move.x, move.y);
      }

      if (animate) {
        var toAttrs = animate.toAttrs,
            animateCfg = __rest(animate, ["toAttrs"]);

        _this.instance.animate(toAttrs, animateCfg);
      }
    };

    _this.bindEvents = function () {
      _this.instance.off();

      (0, _forIn["default"])(_events.EVENTS, function (v, k) {
        if ((0, _isFunction["default"])(_this.props[k])) {
          _this.instance.on(v, _this.props[k]);
        }
      });
    };

    var group = props.group,
        zIndex = props.zIndex,
        name = props.name;
    _this.id = (0, _uniqueId["default"])('group');

    if (group.isChartCanvas) {
      group.chart.on('afterrender', _this.handleRender);
    } else {
      _this.instance = group.addGroup({
        zIndex: zIndex,
        name: name
      });

      _this.configGroup(props);
    }

    return _this;
  }

  (0, _createClass2["default"])(Group, [{
    key: "componentWillUnmount",
    value: function componentWillUnmount() {
      var group = this.props.group;

      if (group.isChartCanvas) {
        group.chart.off('afterrender', this.handleRender);
      }

      if (this.instance) {
        this.instance.remove(true);
      }
    }
  }, {
    key: "getInstance",
    value: function getInstance() {
      return this.instance;
    }
  }, {
    key: "render",
    value: function render() {
      var group = this.props.group;

      if (this.instance) {
        this.instance.clear();
        this.bindEvents();
      }

      return group.isChartCanvas && this.state.isReady || !group.isChartCanvas ? /*#__PURE__*/_react["default"].createElement(_group2["default"].Provider, {
        value: this.instance
      }, /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, {
        key: (0, _uniqueId["default"])(this.id)
      }, this.props.children)) : /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null);
    }
  }]);
  return Group;
}(_react["default"].Component);

Group.defaultProps = {
  zIndex: 3
};

var _default = (0, _group2.withGroupContext)(Group);

exports["default"] = _default;
