// bizcharts/lib/g-components/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Canvas", {
  enumerable: true,
  get: function get() {
    return _Canvas["default"];
  }
});
Object.defineProperty(exports, "Group", {
  enumerable: true,
  get: function get() {
    return _Group["default"];
  }
});
Object.defineProperty(exports, "Circle", {
  enumerable: true,
  get: function get() {
    return _Circle["default"];
  }
});
Object.defineProperty(exports, "Ellipse", {
  enumerable: true,
  get: function get() {
    return _Ellipse["default"];
  }
});
Object.defineProperty(exports, "Image", {
  enumerable: true,
  get: function get() {
    return _Image["default"];
  }
});
Object.defineProperty(exports, "Line", {
  enumerable: true,
  get: function get() {
    return _Line["default"];
  }
});
Object.defineProperty(exports, "Marker", {
  enumerable: true,
  get: function get() {
    return _Marker["default"];
  }
});
Object.defineProperty(exports, "Path", {
  enumerable: true,
  get: function get() {
    return _Path["default"];
  }
});
Object.defineProperty(exports, "Polygon", {
  enumerable: true,
  get: function get() {
    return _Polygon["default"];
  }
});
Object.defineProperty(exports, "Polyline", {
  enumerable: true,
  get: function get() {
    return _Polyline["default"];
  }
});
Object.defineProperty(exports, "Rect", {
  enumerable: true,
  get: function get() {
    return _Rect["default"];
  }
});
Object.defineProperty(exports, "Text", {
  enumerable: true,
  get: function get() {
    return _Text["default"];
  }
});
exports.render = void 0;

var _reconciler = _interopRequireDefault(require("./reconciler"));

var _Canvas = _interopRequireDefault(require("./Canvas"));

var _Group = _interopRequireDefault(require("./Group"));

var _Circle = _interopRequireDefault(require("./Circle"));

var _Ellipse = _interopRequireDefault(require("./Ellipse"));

var _Image = _interopRequireDefault(require("./Image"));

var _Line = _interopRequireDefault(require("./Line"));

var _Marker = _interopRequireDefault(require("./Marker"));

var _Path = _interopRequireDefault(require("./Path"));

var _Polygon = _interopRequireDefault(require("./Polygon"));

var _Polyline = _interopRequireDefault(require("./Polyline"));

var _Rect = _interopRequireDefault(require("./Rect"));

var _Text = _interopRequireDefault(require("./Text"));

var render = _reconciler["default"].render;
exports.render = render;
