// bizcharts/es/geometry/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = _default2;

var _object = _interopRequireDefault(require("object.assign"));

var _react = _interopRequireDefault(require("react"));

var _Area = _interopRequireDefault(require("./Area"));

var _Edge = _interopRequireDefault(require("./Edge"));

var _Heatmap = _interopRequireDefault(require("./Heatmap"));

var _Interval = _interopRequireDefault(require("./Interval"));

var _Line = _interopRequireDefault(require("./Line"));

var _Point = _interopRequireDefault(require("./Point"));

var _Polygon = _interopRequireDefault(require("./Polygon"));

var _LineAdvance = _interopRequireDefault(require("./LineAdvance"));

var _warning = _interopRequireDefault(require("warning"));

var __rest = void 0 && (void 0).__rest || function (s, e) {
  var t = {};

  for (var p in s) {
    if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0) t[p] = s[p];
  }

  if (s != null && typeof Object.getOwnPropertySymbols === "function") for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
    if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i])) t[p[i]] = s[p[i]];
  }
  return t;
};

var GEOM_MAP = {
  area: _Area["default"],
  edge: _Edge["default"],
  heatmap: _Heatmap["default"],
  interval: _Interval["default"],
  line: _Line["default"],
  point: _Point["default"],
  polygon: _Polygon["default"],
  // bx 做了预设的图形
  'line-advance': _LineAdvance["default"]
};

function _default2(props) {
  var type = props.type,
      cfg = __rest(props, ["type"]);

  var Geom = GEOM_MAP[type];
  if (Geom) return /*#__PURE__*/_react["default"].createElement(Geom, (0, _object["default"])({}, cfg));
  (0, _warning["default"])(true, 'Only support the below type: area|edge|heatmap|interval|line|point|polygon|line-advance');
  return null;
}
