// bizcharts/es/boundary/withContainer.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = withContainer;

var _object = _interopRequireDefault(require("object.assign"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

function withContainer(Component) {
  var name = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 'ChartContainer';

  var Cls = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
    var container = (0, _react.useRef)();

    var _useState = (0, _react.useState)(false),
        _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
        inited = _useState2[0],
        setInited = _useState2[1];

    var _props$className = props.className,
        className = _props$className === void 0 ? "bizcharts" : _props$className,
        containerStyle = props.containerStyle;
    (0, _react.useEffect)(function () {
      setInited(true);
    }, []); // @ts-ignore

    return /*#__PURE__*/_react["default"].createElement("div", {
      ref: container,
      className: className,
      // @ts-ignore
      style: (0, _object["default"])({
        position: 'relative',
        height: props.height || '100%',
        width: props.width || '100%'
      }, containerStyle)
    }, inited ? /*#__PURE__*/_react["default"].createElement(Component, (0, _object["default"])({
      ref: ref,
      container: container.current
    }, props)) : /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null));
  });

  Cls.displayName = name || Component.name;
  return Cls;
}

;
